### PR TITLE
JavaScript SDK Documentation Revamp

### PR DESCRIPTION
**Pull request by saikumarrs**
## Description of the change

> Updated JavaScript SDK documentation based on some of the feedback from customers

- Moved content into separate pages (quick start guide, faqs etc)
- Redundant data definitions in SDK were extracted into a separate section "Types"
- Section numbering removed to make it consistent across the docs
- Added a missing section on callback methods
- Made all the style conventions consistent
- Minor issues/typos corrected

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

N/A

### Development

N/A

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by MoumitaM — APPROVED** [review]
LGTM

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/querystring-api.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+http://hostname.com/?ajs_uid=12345&ajs_event=test%20event&ajs_aid=abcde&ajs_prop_testProp=prop1&ajs_trait_name=Firstname+Lastname
+```
+
+The following API calls will be triggered:
+
+```javascript
+rudderanalytics.setAnonymousId("abcde");
+rudderanalytics.identify("12345", { name: "Firstname Lastname" });
+rudderanalytics.track("test event", { testProp: "prop1" });
+```
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk/querystring-api.md:36]: @saikumarrs could you please confirm if the querystring URL + the resulting calls are accurate?

**Comment by ameypv-rudder** [comment]
All set for @arnab-p 's review :)

**Comment by ameypv-rudder** [comment]
As discussed, closing this PR as all the changes are successfully ported to this PR: https://github.com/rudderlabs/rudderstack-docs/pull/249

**Pull request closed without merging**